src/odm_sharing/private/rules.py

```diff
@@ -147,12 +146,14 @@ def get_field_name(column: str) -> str:
         type_class = RULE_FIELD_TYPES[field]
         try:
             typed_val = coerce_value(ctx, type_class, val)
-            result.__setattr__(field, typed_val)
+            object.__setattr__(rule, field, typed_val)
+            assigned_fields.add(field)
         except ParseError as e:
             errors.append(e)
+    assert assigned_fields == RULE_FIELDS, 'some rule fields were not set'
```

Comment:
Is there a situation when this can occur?

Reply:
it's only for debug purposes, it shouldn't happen in production. It's just to prevent mistakes. Should i remove it?

Reply:
I would add a comment saying that's its for debug purposes.